src/services/Aquarius.ts

```diff
@@ -108,111 +108,68 @@ export class Aquarius {
-        LoggerInstance.error(
-          'Metadata validation failed using publisher signature validation (perhaps not supported or legacy Aquarius), retrying with legacy path...',
-          e.message
-        )
-        response = await validateRequestLegacy()
+      const publisherAddress = await signer.getAddress()
+      // aquarius is always same url of other components with ocean nodes
+      const responseNonce = await fetch(pathNonce + `?userAddress=${publisherAddress}`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+        signal
+      })
+      let { nonce } = await responseNonce.json()
+      console.log(`[getNonce] Consumer: ${publisherAddress} nonce: ${nonce}`)
+      if (!nonce || nonce === null) {
+        nonce = '0'
       }
-    } else {
-      // backwards compatibility, "old" way without signature stuff
-      // this will not validate on newer versions of Ocean Node (status:400), as the node will not add the validation signature
-      response = await validateRequestLegacy()
-    }
-    if (!response) return status
-
-    if (response.status === 200) {
-      jsonResponse = jsonResponse || (await response.json())
-      status.valid = true
-      status.hash = jsonResponse.hash
-      status.proof = {
-        validatorAddress: jsonResponse.publicKey,
-        r: jsonResponse.r[0],
-        s: jsonResponse.s[0],
-        v: jsonResponse.v
+      const nextNonce = (Number(nonce) + 1).toString() // have to increase the previous
+      // same signed message as usual (did + nonce)
+      // the node will only validate (add his signature if there fields are present and are valid)
+      // let signatureMessage = publisherAddress
+      const signatureMessage = ddo.id + nextNonce
+      const signature = await signRequest(signer, signatureMessage)
+      const data = { ddo, publisherAddress, nonce: nextNonce, signature }
+      const response = await fetch(ddoValidateRoute, {
+        method: 'POST',
+        body: JSON.stringify(data),
+        headers: { 'Content-Type': 'application/octet-stream' },
+        signal
+      })
+      const jsonResponse = await response.json()
+      if (response.status !== 200) {
+        throw new Error('Metadata validation failed')
       }
-    } else {
-      status.errors = jsonResponse
-      LoggerInstance.error('validate Metadata failed:', response.status, status.errors)
+      console.log('Ddo successfully validated')
+
+      return {
+        valid: true,
+        hash: jsonResponse.hash,
+        proof: {
+          validatorAddress: jsonResponse.publicKey,
+          r: jsonResponse.r[0],
+          s: jsonResponse.s[0],
+          v: jsonResponse.v
+        }
+      }
+    } catch (e) {
+      // retry with legacy path validation
+      LoggerInstance.error(
+        'Metadata validation failed using publisher signature validation (perhaps not supported or legacy Aquarius), retrying with legacy path...',
```

Comment:
There is no retry for legacy Aquarius, if we want backwards compatibility, then we should retry the legacy path, otherwise only the log should be updated.